engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingwavg/BaseBigNumberRollingWAvgOperator.java

```diff
@@ -0,0 +1,214 @@
+    @NotNull
+    final MathContext mathContext;
+    final String weightColumnName;
+    final ChunkType weightChunkType;
+    final Class<?> weightColumnSourceType;
+
+    protected abstract class Context extends BaseObjectUpdateByOperator<BigDecimal>.Context {
+        protected ObjectChunk<T, ? extends Values> influencerValuesChunk;
+        protected WritableObjectChunk<BigDecimal, ? extends Values> influencerWeightValuesChunk;
+        protected AggregatingObjectRingBuffer<BigDecimal> windowValues;
+        protected AggregatingObjectRingBuffer<BigDecimal> windowWeightValues;
+
+        protected Context(final int affectedChunkSize, final int influencerChunkSize) {
+            super(affectedChunkSize);
+            windowValues = new AggregatingObjectRingBuffer<>(BUFFER_INITIAL_CAPACITY, BigDecimal.ZERO,
+                    BigDecimal::add,
+                    (a, b) -> {
+                        if (a == null && b == null) {
+                            // Return the identity value.
+                            return BigDecimal.ZERO;
+                        } else if (a == null) {
+                            return b;
+                        } else if (b == null) {
+                            return a;
+                        }
+                        return a.add(b);
+                    });
+            windowWeightValues = new AggregatingObjectRingBuffer<>(BUFFER_INITIAL_CAPACITY, BigDecimal.ZERO,
+                    BigDecimal::add,
+                    (a, b) -> {
+                        if (a == null && b == null) {
+                            // Return the identity value.
+                            return BigDecimal.ZERO;
+                        } else if (a == null) {
+                            return b;
+                        } else if (b == null) {
+                            return a;
+                        }
+                        return a.add(b);
+                    });
+            influencerWeightValuesChunk = WritableObjectChunk.makeWritableChunk(influencerChunkSize);
+        }
+
+        @Override
+        public void close() {
+            super.close();
+            windowValues = null;
+            windowWeightValues = null;
+            influencerWeightValuesChunk.close();
+            influencerWeightValuesChunk = null;
+        }
+
+        @Override
+        public void setValueChunks(@NotNull final Chunk<? extends Values>[] valueChunks) {
+            influencerValuesChunk = valueChunks[0].asObjectChunk();
+
+            // Upcast the weight values to BigDecimal
+            switch (weightChunkType) {
+                case Char:
+                    final CharChunk<?> charChunk = valueChunks[1].asCharChunk();
+                    for (int ii = 0; ii < valueChunks[1].size(); ii++) {
+                        final char val = charChunk.get(ii);
+                        influencerWeightValuesChunk.set(ii, Basic.isNull(val) ? null : BigDecimal.valueOf(val));
```

Comment:
I don't think we should use `Basic.isNull` here. This forces the primitive to be boxed. Can we compare against `QueryConstants.NULL_TYPE`  values instead?